### PR TITLE
Non-ingesters should not fail on its config validation

### LINKED ISSUE #7822: Non-ingesters fail to validate config with spread-minimizing token generation strategy
#### Describe the bug
Non-ingester components fail to validate mimir config if `ingester.ring.token_generation_strategy` is set to `spread-minimizing`:
```
error validating config: invalid ingester config: unable to extract instance id from "mimir-query-scheduler-55f8bb8f9b-48pgb"
```

#### To Reproduce

Helm chart version: 5.2.1 (mimir-distributed)
Mimir version: 2.11

Add the following to the chart values:
```
mimir:
  structuredConfig:
    ingester:
      ring:
        tokens_file_path: ''
        token_generation_strategy: spread-minimizing
        spread_minimizing_zones: az-1,az-2,az-3
        spread_minimizing_join_ring_in_order: true
```

All components, except ingester, will start crashing on startup with:
```
error validating config: invalid ingester config: unable to extract instance id from "mimir-query-scheduler-55f8bb8f9b-48pgb"
```

#### Expected behavior

Non-ingester components should ignore instance id as this part of config is not relevant to them.

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: helm

### PR DESCRIPTION
**Pull request by narqo**
#### What this PR does

These changes fix an issue with mimir's config validation, where non-ingester modules fail on its config validation, when `-ingester.ring.token-generation-strategy=spread-minimizing` is set.

#### Which issue(s) this PR fixes or relates to

Fixes #7822

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/mimir/mimir_test.go [hunk 1]
```diff
@@ -464,6 +464,18 @@ func TestConfigValidation(t *testing.T) {
 			},
 			expectAnyError: true,
 		},
+		{
+			name: "should pass if ingester ring has token generation strategy set to spread-minimizing, and target is not ingester",
+			getTestConfig: func() *Config {
+				cfg := newDefaultConfig()
+				_ = cfg.Target.Set("distributor")
```
Comment (lamida) [inline pkg/mimir/mimir_test.go:471]: Should we add test for ingester that will throw error for invalid zones naming when spead-minimizing is enabled?

**Review by lamida — APPROVED** [review]
Added a comment about test, but in general LGTM.

File: pkg/mimir/mimir.go [hunk 2]
```diff
@@ -267,8 +267,13 @@ func (c *Config) Validate(log log.Logger) error {
 	if err := c.IngesterClient.Validate(log); err != nil {
 		return errors.Wrap(err, "invalid ingester_client config")
 	}
-	if err := c.Ingester.Validate(log); err != nil {
-		return errors.Wrap(err, "invalid ingester config")
+	// We check for "ingester" module here because, as of today, its config has a special mode, that assumes
+	// passing a unique set of per instance flags, e.g. "-ingester.ring.instance-id". Such scenario breaks
+	// the validation of other modules, if those flags aren't also passed to each instance (ref grafana/mimir#7822).
+	if c.isAnyModuleEnabled(Ingester, Write, All) {
+		if err := c.Ingester.Validate(log); err != nil {
```
[the remark this answers is not in the document]
Comment (narqo) [reply pkg/mimir/mimir.go]: We can do that, indeed. PHAL

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM! There's a conflict in the changelog. Can you resolve that and I can merge the PR?

Thanks for the cleanup btw :)

File: pkg/mimir/mimir.go [hunk 3]
```diff
@@ -268,7 +268,14 @@ func (c *Config) Validate(log log.Logger) error {
...
 	if err := c.Ingester.Validate(log); err != nil {
-		return errors.Wrap(err, "invalid ingester config")
+		// We check for "ingester" module here because, as of today, its config has a special mode, that assumes
+		// passing a unique set of per instance flags, e.g. "-ingester.ring.instance-id".
+		// Such a scenario breaks the validation of other modules if those flags aren't also passed to each instance (ref
+		// grafana/mimir#7822). Otherwise, log the warning and move on.
+		if c.isAnyModuleEnabled(Ingester, Write, All) || !errors.Is(err, ingester.ErrSpreadMinimizingValidation) {
+			return errors.Wrap(err, "invalid ingester config")
+		}
+		level.Warn(log).Log("ingester config is invalid; moving on because the \"ingester\" module is not in this process's targets", "err", err.Error())
```
Comment (dimitarvdimitrov) [inline pkg/mimir/mimir.go:278]: i'd opt for a debug log here. It shouldn't be anything to worry about. WDYT?

**Review by dimitarvdimitrov — APPROVED** [review]

**Pull request merged**